crates/storage/codecs/src/alloy/withdrawal.rs

```diff
@@ -52,6 +53,52 @@ impl Compact for AlloyWithdrawal {
     }
 }
 
+/// Represents a collection of [`Withdrawals`].
+#[derive(Debug, Clone, PartialEq, Eq, Default, Compact)]
+#[cfg_attr(
+    any(test, feature = "test-utils"),
+    derive(arbitrary::Arbitrary, serde::Serialize, serde::Deserialize)
+)]
+#[cfg_attr(feature = "test-utils", allow(unreachable_pub), visibility::make(pub))]
+#[add_arbitrary_tests(compact)]
+pub(crate) struct Withdrawals(Vec<Withdrawal>);
```

Comment:
@joshieDo this doesn't seem necessary, because this should just be the Vec impl

```rust
impl Compact for Withdrawals {
        fn to_compact<B>(&self, buf: &mut B) -> usize
        where
            B: bytes::BufMut + AsMut<[u8]>,
        {
            let mut flags = WithdrawalsFlags::default();
            let mut total_length = 0;
            let mut buffer = bytes::BytesMut::new();
            let _len = self.0.to_compact(&mut buffer);
            let flags = flags.into_bytes();
            total_length += flags.len() + buffer.len();
            buf.put_slice(&flags);
            buf.put(buffer);
            total_length
        }
        fn from_compact(mut buf: &[u8], len: usize) -> (Self, &[u8]) {
            let (flags, mut buf) = WithdrawalsFlags::from(buf);
            let (placeholder, new_buf) = Vec::from_compact(buf, buf.len());
            buf = new_buf;
            let obj = Withdrawals(placeholder);
            (obj, buf)
        }
    }


   impl WithdrawalsFlags {
        /// Placeholder: does not read any value.
        pub fn from(mut buf: &[u8]) -> (Self, &[u8]) {
            (WithdrawalsFlags::default(), buf)
        }
        /// Placeholder: returns an empty array.
        pub fn into_bytes(self) -> [u8; 0] {
            []
        }
    }
``` 

so we should be able to implement this directly without doing a collect

see also:
https://github.com/paradigmxyz/reth/blob/main/crates/storage/codecs/src/alloy/access_list.rs#L31-L31

Reply:
@mattsse I think I get your point here. If I'm not mistaken, you want to use the `Compact` implementation for `Vec<T>` with `T` being `Withdrawal` for which `Compact` is already implemented above (in order to avoid the collect). Indeed, I looked at what is done similarly for `AccessList` which is a vector of `AccessListItem` and so according to your comment we could simply do something similar for `Withdrawals` without the declaration:

```rust
pub(crate) struct Withdrawals(Vec<Withdrawal>);
```

I implemented in my last push the `from_compact` using `Vec::from_compact`. However I have some remarks:
- If we no longer declare `Withdrawals` structures in codecs::alloy here then here

https://github.com/paradigmxyz/reth/blob/2e750f0ca03043cd2370036cc4ed7c9e1c12f173/crates/cli/commands/src/test_vectors/compact.rs#L78

we will no longer be able to create the vector for the tests right?

- The implementation of `to_compact` would look like

```rust
impl Compact for AlloyWithdrawals {
fn to_compact<B>(&self, buf: &mut B) -> usize
where
B: bytes::BufMut + AsMut<[u8]>,
{
let mut buffer = Vec::new();
self.0.to_compact(&mut buffer);
buf.put(&buffer[..]);
buffer.len()
}
```

but in this case we need to access `self.0` which is not currently public for `Withdrawals` https://github.com/alloy-rs/alloy/blob/b227b17ac08f966be3459c1776bb450d15612fcb/crates/eips/src/eip4895.rs#L60

Depending on the criticality of this issue we could maybe merge as is and open a follow-up issue to fix the collect issue in this implementation, what do you think?

Reply:
right, we should definitely expose a function to obtain a slice of the wrapper vec on withdrawals

Reply:
we can also do this

Reply:
@mattsse Done here https://github.com/alloy-rs/alloy/pull/1590.

Maybe, depending on the criticality of this adjustment here, we could merge this PR to continue moving forward and track the `to_compact` change in a follow up issue, what do you think?